### PR TITLE
feat: full gravity multichain support

### PR DESCRIPTION
**Pull request by byte-bandit**
# Related Github tickets

- https://github.com/VolumeFi/paloma/issues/1636
- https://github.com/VolumeFi/paloma/issues/842
- https://github.com/VolumeFi/paloma/issues/1378
- https://github.com/VolumeFi/paloma/issues/1617

# Background

This change addresses quite a lot of ongoing issues on the gravity project, and the merge will result in a fully functional bridge implementation across multiple chains and tokens, with full support for funds exchange.

Rollout will require an updated version of compass with this change https://github.com/VolumeFi/paloma-compass-evm/commit/95b856db6d3a7f0d608a75e9527bab086b587bad

# Testing completed

- [x] test coverage exists or has been added/updated
- [x] tested in a private testnet

# Breaking changes

- [x] I have checked my code for breaking changes
- [x] If there are breaking changes, there is a supporting migration.

### CODE REVIEW

File: x/evm/keeper/smart_contract_deployment.go [hunk 1]
```diff
@@ -247,7 +247,7 @@ func (k Keeper) deploySmartContractToChain(ctx context.Context, chainInfo *types
 	}
 	uniqueID := generateSmartContractID(ctx)
 	k.createSmartContractDeployment(ctx, smartContract, chainInfo, uniqueID[:])
-	lastEventNonce, err := k.Gravity.GetLastObservedEventNonce(ctx, chainInfo.GetChainReferenceID())
+	lastEventNonce, err := k.Gravity.GetLastObservedGravityNonce(ctx, chainInfo.GetChainReferenceID())
 	if err != nil {
 		return fmt.Errorf("failed to get last observed event nonce: %w", err)
```
Comment (maharifu) [inline x/evm/keeper/smart_contract_deployment.go:252]: Should we rename these to `lastGravityNonce` to keep consistency?
```suggestion
	lastGravityNonce, err := k.Gravity.GetLastObservedGravityNonce(ctx, chainInfo.GetChainReferenceID())
	if err != nil {
		return fmt.Errorf("failed to get last observed gravity nonce: %w", err)
```

**Review by maharifu — COMMENTED** [review]
LGTM, just one small nitpick.

**Review by maharifu — APPROVED** [review]
Nice one :+1:

**Comment by taariq** [comment]
@byte-bandit @maharifu `The base branch requires all commits to be signed` blocking merging.

**Review by taariq — APPROVED** [review]
LFG!

**Pull request merged**